app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -0,0 +1,40 @@
+module OpenidConnect
+  class AuthorizationController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def index
+      @authorize_form = OpenidConnectAuthorizeForm.new(params)
+
+      success = @authorize_form.valid?
+      analytics.track_event(Analytics::OPENID_CONNECT_REQUEST_AUTHORIZATION,
+                            success: success,
+                            client_id: @authorize_form.client_id,
+                            errors: @authorize_form.errors.full_messages)
+      if success
+        render :index
+      else
+        render :error
+      end
+    end
+
+    def create
+      @authorize_form = OpenidConnectAuthorizeForm.new(params)
+      redirect_uri = @authorize_form.submit(current_user)
+
+      if redirect_uri
+        analytics.track_event(Analytics::OPENID_CONNECT_ALLOW, client_id: @authorize_form.client_id)
+        redirect_to redirect_uri
+        return
+      end
```

Comment:
Correct me if I'm wrong. It looks like `@authorize_form.submit` will always return a URI unless there is no `redirect_uri` in the `params`. If the `redirect_uri` is not valid, do we still want to capture the event as `OPENID_CONNECT_ALLOW`, which implies success?

Should we instead stick to our convention of having the form return a hash with `success` and `errors` and respond accordingly, and track analytics with that rich hash instead of just the client_id?